orchestra/static/orchestra/task/js/required_fields.js

```diff
@@ -0,0 +1,90 @@
+(function() {
+  'use strict';
+
+  var serviceModule =  angular.module('orchestra.task.services');
+
+  /**
+   * Service to keep track of all required fields in a task view and
+   * validate them on submit.
+   */
+  serviceModule.factory('requiredFields', function($rootScope, orchestraService) {
+    var requiredFields = {
+      fields: [],
+      invalid: [],
+      setup: function(data) {
```

Comment:
This function was originally being called by custom task view controllers, but I think it makes sense to set this up in the base task controller instead.

Data is the base object which field indicators correspond to. I'll add a comment.